### PR TITLE
feat(test-command): add @sasjs/server support

### LINKED ISSUE #1108: sasjs test on sasjs/server
We need to be able to run `sasjs test` on [sasjs/server](https://github.com/sasjs/server)!

As a pre-requisite, we would also need sasjs request.

And given how similar it is, we may as well implement sasjs job execute support at the same time.

### PR DESCRIPTION
**Pull request by YuryShkoda**
## Issue

Closes #1108 

## Intent

- Add `@sasjs/server` support to `sasjs test` command

## Implementation

- Added logic to support `SASJS` `ServerType`.
- Added unit test to cover the new functionality.

## Checks

- [x] Code is formatted correctly (`npm run lint:fix`).
image
- [x] Any new functionality has been unit tested.
- [x] All unit tests are passing (`npm test`).
Mocked:
image
Server (without `request.spec.server`):
image
- [ ] All CI checks are green.
- [ ] JSDoc comments have been added or updated.
- [x] Reviewer is assigned.

### CODE REVIEW

File: src/commands/testing/spec/testing.spec.server.ts [hunk 1]
```diff
@@ -328,152 +97,392 @@ BRF:1
...
 end_of_record`
 
-    const coverageLcov = await readFile(coverageLcovPath)
+  describe('SASVIYA', () => {
+    const target: Target = generateTarget()
+    const testUrl = (test: string) =>
+      `${target.serverUrl}/${
+        target.serverType === ServerType.SasViya
+          ? 'SASJobExecution'
+          : 'SASStoredProcess'
```
Comment (allanbowe) [inline src/commands/testing/spec/testing.spec.server.ts:106]: is this VIYA specific?  If so I guess it doesn't need SASStoredProcess.  If it's generic, it should probably include the Server API route as well?

File: src/commands/testing/spec/testing.spec.server.ts [hunk 2]
```diff
@@ -497,13 +506,313 @@ testteardown,tests/testteardown.sas,sasjs_test_id,not provided,,${testUrlLink(
...
+            '{"test_results":\r\n' +
+            '[\r\n' +
+            '{"TEST_DESCRIPTION":"random test description" ,"TEST_RESULT":"PASS" }\r\n' +
+            ']}',
+          log:
+            '1                                          The SAS System             11:42 Friday, February 11, 2022\r\n' +
+            '\r\n' +
+            'NOTE: Copyright (c) 2016 by SAS Institute Inc., Cary, NC, USA. \r\n' +
+            'NOTE: SAS (r) Proprietary Software 9.4 (TS1M6) \r\n' +
+            '      Licensed to ANALYTIUM LTD- PARTNER LICENCE DATA MANAGER CLIENT, Site 70221619.\r\n' +
```
Comment (allanbowe) [inline src/commands/testing/spec/testing.spec.server.ts:714]: This part will break if we ever use a different platform

Comment (allanbowe) [reply src/commands/testing/spec/testing.spec.server.ts:714]: The date will also be different whenever SAS restarts

Comment (allanbowe) [reply src/commands/testing/spec/testing.spec.server.ts:714]: Ignore these comments - I did not realise this was a mocked response

**Review by allanbowe — APPROVED** [review]

**Review by VladislavParhomchik — DISMISSED** [review]

**Review by allanbowe — CHANGES_REQUESTED** [review]
This is working fine (and SUPER fast!!), the only issue I see is that the console log path says `/SASStoredProcess` when it should say `/SASjsApi/stp/execute`.

This is just a display issue (also a functionality issue as you can't click it).  The test does indeed run.

image

**Review by allanbowe — APPROVED** [review]

**Pull request merged**